mixer/test/client/istio_authn_no_bound_origin/authn_report_test.go

```diff
@@ -0,0 +1,183 @@
+const reportAttributesOkGet = `
+{
+  "context.protocol": "http",
+  "mesh1.ip": "[1 1 1 1]",
+  "mesh2.ip": "[0 0 0 0 0 0 0 0 0 0 255 255 204 152 189 116]",
+  "mesh3.ip": "[0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 8]",
+  "request.host": "*",
+  "request.path": "/echo",
+  "request.time": "*",
+  "request.useragent": "Go-http-client/1.1",
+  "request.method": "GET",
+  "request.scheme": "http",
+  "source.uid": "POD11",
+  "source.namespace": "XYZ11",
+  "source.ip": "[127 0 0 1]",
+  "source.port": "*",
+  "destination.ip": "[127 0 0 1]",
+  "destination.port": "*",
+  "target.name": "target-name",
+  "target.user": "target-user",
+  "target.uid": "POD222",
+  "target.namespace": "XYZ222",
+  "connection.mtls": false,
+  "check.cache_hit": false,
+  "quota.cache_hit": false,
+  "request.headers": {
+     ":method": "GET",
+     ":path": "/echo",
+     ":authority": "*",
+     "x-forwarded-proto": "http",
+     "x-istio-attributes": "-",
+     "x-request-id": "*"
+  },
+  "request.size": 0,
+  "response.time": "*",
+  "response.size": 0,
+  "response.duration": "*",
+  "response.code": 200,
+  "response.headers": {
+     "date": "*",
+     "content-length": "0",
+     ":status": "200",
+     "server": "envoy"
+  },
+  "request.auth.audiences": "aud1",
+  "request.auth.claims": {
+     "iss": "issuer@foo.com", 
+     "sub": "sub@foo.com",
+     "aud": "aud1",
+     "some-other-string-claims": "some-claims-kept"
+  }
+}
+`
+
+// The Istio authn envoy config
+const authnConfig = `
+{
+  "type": "decoder",
+  "name": "istio_authn",
+  "config": {
+    "policy": {
+      "origins": [
+        {
+          "jwt": {
+            "issuer": "issuer@foo.com",
+            "jwks_uri": "http://localhost:8081/"
+          }
+        }
+      ]
+    },
+    "jwt_output_payload_locations": {
+      "issuer@foo.com": "sec-istio-auth-jwt-output"
+    }
+  }
+},
+`
+
+const secIstioAuthUserInfoHeaderKey = "sec-istio-auth-jwt-output"
+
+const secIstioAuthUserinfoHeaderValue = `
+{
+  "iss": "issuer@foo.com",
+  "sub": "sub@foo.com",
+  "aud": "aud1",
+  "non-string-will-be-ignored": 1512754205,
+  "some-other-string-claims": "some-claims-kept"
+}
+`
+
+func TestAuthnCheckReportAttributesNoBoundToOrigin(t *testing.T) {
+	// In the Envoy config, no binding to origin
+	s := env.NewTestSetupWithEnvoyConfig(env.CheckReportIstioAuthnAttributesTestNoBoundToOrigin,
+		authnConfig, t)
+
+	env.SetStatsUpdateInterval(s.MfConfig(), 1)
+	if err := s.SetUp(); err != nil {
+		t.Fatalf("Failed to setup test: %v", err)
+	}
+	defer s.TearDown()
+
+	url := fmt.Sprintf("http://localhost:%d/echo", s.Ports().ClientProxyPort)
+
+	// Issues a GET echo request with 0 size body
+	tag := "OKGet"
+
+	// Add jwt_auth header to be consumed by Istio authn filter
+	headers := map[string]string{}
+	headers[secIstioAuthUserInfoHeaderKey] =
+		base64.StdEncoding.EncodeToString([]byte(secIstioAuthUserinfoHeaderValue))
+
+	if _, _, err := env.HTTPGetWithHeaders(url, headers); err != nil {
```

Comment:
please add a test to test rejected requests by authn

Reply:
Done: added two tests to test rejected requests by authn.